doc/extensions/acrbuildext/azext_acrbuildext/build_task.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import LongRunningOperation
+from ._utils import (
+    arm_deploy_template_build_task_create,
+    validate_managed_registry
+)
+from .build import _check_image_name
+
+BUILD_TASKS_NOT_SUPPORTED = 'Build Tasks are only supported for managed registries.'
+
+
+def acr_build_task_create(
+    cmd,
+    client,
+    build_task_name,
+    registry_name,
+    context,            
+    source_branch,  
+    image_name,
+    git_access_token,
+    resource_group_name=None,
```

Comment:
I think we can remove resource_group_name @djyou

Reply:
No, it's just being compatible with Az commands in general. `resource_group_name` is a required parameter and people who are familiar with other command modules will most likely specify a `resource_group_name` when first working with acr.

Perf wise, specifying a `resource_group_name` means the CLI can avoid an additional query (with possible pagination) to find out what resource group is.

Reply:
Yes that's why `resource_group_name` is optional so people don't have to type it and your scenario is not affected, I believe design-wise we are on the same page. But we also don't want to fail people who are used to non-acr commands and who type a `resource_group_name` because that's what other command modules require. It's more of a compatibility or consistency consideration vs. acr is standing out and being different from others. We agreed to make it optional in the past.

That said, I am not in a position to stop you from removing it, although I do agree with consistency consideration.

Perf is just a small price we have to pay to support optional `resource_group_name`. I don't think it'll be a concern.